SendGrid -> Nodemailer

13circle opened the pull request:
resolve https://github.com/mju-likelion/antigua/issues/9

13circle committed:
Chore: Add nodemailer package

13circle committed:
Refactor: Apply nodemailer instead of sendgrid

13circle committed:
Fix: Send approval notification mail to admin

noah0316 commented on src/models/user.ts:
```diff
@@ -105,85 +105,134 @@ UserSchema.methods.generateToken = function (): boolean | string {
-    await sgMail.send(msg);
-    console.log('send mail success!');
+    await transporter.sendMail(msg);
+    console.log('send mail, success!');
```
consola 적용은 어떠하신지!!

13circle replied:
console.success 하면 됨?

noah0316 replied:
https://github.com/mju-likelion/antigua/pull/6#discussion_r559092057
요거 한번 참고하시면 될 것 같아요 ㅎㅎ 간단한 사용법 적어놓았습니다 :)

13circle committed:
Fix: Apply consola to log success

ndaemy commented on package.json:
```diff
@@ -14,7 +14,8 @@
     "koa-router": "^9.4.0",
-    "mongoose": "^5.10.6"
+    "mongoose": "^5.10.6",
+    "nodemailer": "^6.4.17"
```
위의 @sendgrid/mail은 삭제해야 할듯

13circle replied:
나루호도

13circle committed:
Chore: Remove @sendgrid/mail package

ndaemy commented on src/models/user.ts:
```diff
@@ -105,85 +105,134 @@ UserSchema.methods.generateToken = function (): boolean | string {
+    MAIL_SENDER_IS_SECURE,
+    MAIL_SENDER_EMAIL,
+    MAIL_SENDER_PASSWORD,
+  } = process.env;
```
이렇게 받으면, MAIL_SENDER_*는 string | undefined 입니다. 그래서 undefined에 대한 예외 처리를 해주던가, 아니면 118~124번째 줄에 default value를 넣어줘야 에러가 안나요.

13circle replied:
에러가 난다는게 예외처리를 안 해줬을 때 난다는 말씀이시죠?

ndaemy replied:
예외처리 안함 && process.env.MAIN_SENDER_*가 없음
이면 에러납니다

13circle committed:
Fix: Handle env vars exceptions for unexistance

13circle committed:
Refactor: Apply OAuth2 Gmail sender

- Create project at GCP
- Apply Gmail API
- Create OAuth2 credentials in Gmail API
- Connect the credentials with Google OAuth2.0 Playground
- Add OAuth2 credentials & refresh token to .env
- Bring added credentials & refresh token & configure nodemailer transporter

- Reference:
https://stackoverflow.com/questions/24098461/nodemailer-gmail-what-exactly-is-a-refresh-token-and-how-do-i-get-one/24123550
